rllib/execution/parallel_requests.py

```diff
@@ -101,20 +110,19 @@ def asynchronous_parallel_requests(
             len(remote_requests_in_flight[actor])
             < max_remote_requests_in_flight_per_actor
         ):
-            if remote_fn is None:
-                req = actor.sample.remote()
-            else:
+            if remote_fn is not None:
```

Comment:
Should we nix this if-block? I don't think `remote_fn` can be None anymore as we now have a better default value in the function's signature, correct?

Reply:
we need remote_fn is none for the behavior where we are doing our large ray.wait call on the requests in flight